File: front-end/src/domain/entities/Gender.ts
export class Gender {
    constructor(
        private readonly _id: number,
        private _name: string
    ){
        this._id = _id;
        this._name = _name;
    }

    get id(){
        return this._id;
    }

    get name(){
        return this._name;
    }

    setName(name: string){
        this._name = name;
    }
}

File: back-end/src/domain/entities/gender.rs
use serde::{Serialize, Deserialize};
use uuid::Uuid;


#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
pub struct Gender {
    id: Uuid,
    name: String
}


impl Gender {
    pub fn new(id: Uuid, name: String) -> Gender {
        Gender { id, name }
    }

    pub fn get_id(&self) -> Uuid {
        self.id
    }

    pub fn get_name(&self) -> String {
        self.name.clone()
    }

    pub fn set_name(&mut self, new_name: String) {
        self.name = new_name;
    }
}